app/assets/javascripts/merge_status.coffee

```diff
@@ -42,6 +42,10 @@ class MergeStatusPoller
         container.innerHTML = html
         @onPageChange()
 
+  reloadUrl: =>
+    mergeForm =  document.querySelector('form#merge-form')
```

Comment:
Ok, so you are going to hate me because I told you to do this.

Turns out the form contains these 3 hidden fields:

```
 <input name="utf8" type="hidden" value="✓">
 <input type="hidden" name="_method" value="put">
 <input type="hidden" name="authenticity_token" value="[REDACTED]">
```

The first two we could almost ignore, it's just ugly but fine.

But the 3rd one is different on every request, which mean we'd totally  break [the HTTP caching](https://github.com/Shopify/shipit-engine/blob/ff515fd8142b8d2d4b5fe7d5c83b5a67add52f50/app/controllers/shipit/merge_status_controller.rb#L14).

Also that same caching might cause you troubles, you'll need to include any form attribute you want to persist in there.

Really the HCTW iframe is not well suited for stateful inputs :/

I honestly have no idea how to implement that without lots of warts.

Maybe that data shouldn't be included in the URL, but reset by the javascript in the `fetchPage ` callback?

Reply:
So I didn't go through your suggestion fully, I'm using this to serialize the forum state into the query string but I'm only using the single `unsafe_to_rollback` value to seed the initial checkbox state by. This is the page that the refresh mechanism polls, I'm not manipulating the url at all for the original page. 

The rest of the query params are actually just ignored, including the authenticity token.

&yeah I totally found that this wasn't meant for stateful inputs. Hopefully this isn't too nasty of a workaround though.